lib/ansible/modules/cloud/amazon/ec2_metadata_facts.py

```diff
@@ -445,7 +446,16 @@ def __init__(self, module, ec2_metadata_uri=None, ec2_sshdata_uri=None, ec2_user
         self._prefix = 'ansible_ec2_%s'
 
     def _fetch(self, url):
-        (response, info) = fetch_url(self.module, url, force=True)
+        response, info = fetch_url(self.module, url, force=True)
+
+        if not info.get('msg', '').startswith('OK'):
```

Comment:
Just tested this and it might not be enough to check for just `'OK'`; when the recursive URL calls get to the end of the public key URL it will return a 404. 
```
{'body': '<?xml version="1.0" encoding="iso-8859-1"?>\n<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0
Transitional//EN"\n         "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">\n<html xmlns="http://www.w3.org/1999/xhtml"
xml:lang="en" lang="en">\n <head>\n  <title>404 - Not Found</title>\n </head>\n <body>\n  <h1>404 - Not Found</h1>\n
</body>\n</html>\n', 'date': 'Wed, 18 Apr 2018 18:21:46 GMT', 'content-length': '345', 'status': 404, 'msg': 'HTTP Error 404: Not
Found', 'url': u'http://169.254.169.254/latest/meta-data/public-keys/0=vinay.dandekar', 'content-type': 'text/html', 'connection':
'close', 'server': 'EC2ws'}
```
We could attempt to exclude that? Or maybe for something like: 
```python
if info.get('status') not in [200, 404]:
```